rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,1001 @@
+
+#### Best practices for dynamic contents:  <!-- omit in toc -->
+
+- Dynamic content which updates without a page reload should either be a live region or a widget.
+- Use `aria-live` or [roles with implicit live region attributes](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#roles_with_implicit_live_region_attributes) for dynamic content.
+- Make sure to use `POLITENESS_SETTING` for `aria-live` depending on the criticality of the content.
+
+**aria-live**
+
+`aria-live` attribute can be used to make a live region container. It accepts three values:
+
+| Value           | Description                                                                                                                                                                    |
+| :-------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `assertive`     | Indicates that updates to the region have the highest priority and should be presented the user immediately.                                                                   |
+| `off (default)` | Indicates that updates to the region should not be presented to the user unless the user is currently focused on that region.                                                  |
+| `polite`        | Indicates that updates to the region should be presented at the next graceful opportunity, such as at the end of speaking the current sentence or when the user pauses typing. |
+
+#### Platform Specific Implementation <!-- omit in toc -->
+
+**Web:**
+
+In web we can simply add `aria-live` to any element or use [roles which have implicit live region attributes](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#roles_with_implicit_live_region_attributes)
+
+> [Demo](https://codesandbox.io/s/blade-a11y-rfc-arialive-demo-h53z61?file=/src/App.tsx)
+
+https://user-images.githubusercontent.com/35374649/170882365-47b9caa8-eff8-4572-a319-d0f63785f497.mov
+
+We can also use the [@react-aria/live-announcer](https://github.com/adobe/react-spectrum/tree/main/packages/%40react-aria/live-announcer) package which gives us programmatic way of triggering announcements with `announce()` function, by creating temporary live-regions.
+
+**Native:**
+
+In react-native we have a [announceForAccessibility](https://reactnative.dev/docs/accessibilityinfo#announceforaccessibility) API in AccessibilityInfo interface for announcing dynamic content:
+
+> [Demo](https://snack.expo.dev/K17t0cAtLj)
+
+```ts
+React.useEffect(() => {
+  AccessibilityInfo.announceForAccessibility('Some content to be announced');
+}, []);
+```
+
+https://user-images.githubusercontent.com/35374649/170883127-b81bc324-3d5c-4b9d-905f-14733fcc821d.mp4
+
+
+### ARIA Patterns
```

Comment:
this section doesn't seems like fitting under Screen readers, feels like this is more generic?

Reply:
This is kind of fits, I don't know where to put it. It will just hang out in the middle of the document otherwise.

Reply:
it can become it's own topic maybe while we are talking about ARIA things since this is setting the context and basic information.

Reply:
Made it it's own section instead of subsection of screen readers. makes more sense.